### PR TITLE
Add a lint rule to catch console logging

### PR DESCRIPTION
**Pull request by Scy-D2L**
- Lint fails when code contains `console.log(...)`
- Still allows `console.warn()` and `console.error()`
- This rule ignores our demo pages where we're often console logging on purpose
- Has an exception for getLocalizeResources which seems to be logging deliberately only on debug

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -4,6 +4,15 @@
     "Prism": false
   },
   "rules": {
-    "linebreak-style": ["error", "unix"]
-  }
+    "linebreak-style": ["error", "unix"],
+    "no-console": ["error", { "allow": ["warn", "error"] }]
```
Comment (dbatiste) [inline .eslintrc.json:8]: For this part, I think we need to fix [the config that's being extended](https://github.com/Brightspace/eslint-config-brightspace/blob/master/index.js).

File: .eslintrc.json [hunk 2]
```diff
@@ -4,6 +4,15 @@
...
+  },
+  "overrides": [{
+    "files": [
+      "**/demo/**.html",
```
Comment (dbatiste) [inline .eslintrc.json:12]: This is pretty cool - I did not know we could do this.  :)

File: .eslintrc.json (hunk 1 above)
Comment (dbatiste) [reply .eslintrc.json:8]: That is, I think `eslint:recommended` ruleset changed within es-lint, and we didn't realize.

Comment (dlockhart) [reply .eslintrc.json:8]: Agreed, and as long as we do a minor release because it's still `<0` it'll be treated like a major one.

**Pull request merged**